File: prepper-bunker-backend/src/main/java/org/jonasfroeller/model/Weapon.java
package org.jonasfroeller.model;

import jakarta.persistence.*;

@Entity
public class Weapon {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;
    public String type;
    public String model;
    public int quantity;

    @ManyToOne
    @JoinColumn(name = "ammunition_type_id")
    public AmmunitionType ammunitionType;

    @ManyToOne
    @JoinColumn(name = "storage_location_id")
    public StorageLocation storageLocation;
}

File: prepper-bunker-frontend/src/app/models/weapon.ts
export interface Weapon {
  id: number;
  type: string;
  model: string;
  quantity: number;
  ammunitionTypeId: number;
  storageLocationId: number;
}

export interface WeaponCreate {
  type: string;
  model: string;
  quantity: number;
  ammunitionTypeId: number;
  storageLocationId: number;
}
